forch/simple_auth_state_machine.py

```diff
@@ -1,39 +1,95 @@
 """Module that implements MAB state machine"""""
 
 import logging
+from threading import Lock
+import time
 
 LOGGER = logging.getLogger('mabsm')
 
 
 class AuthStateMachine():
     """Class represents the state machine that handles the Auth for a session"""""
 
+    UNAUTH = "Unauthorized"
+    REQUEST = "RADIUS Request"
+    ACCEPT = "Authorized"
+    MAX_RADIUS_RETRIES = 5
+    RADIUS_QUERY_TIMEOUT = 10
+    REJECT_STATE_TIMEOUT = 20
+    AUTH_SESSION_TIMEOUT = 20
+
     def __init__(self, src_mac, port_id, radius_query_callback, auth_callback):
         self.src_mac = src_mac
         self.port_id = port_id
         self.auth_callback = auth_callback
         self.radius_query_callback = radius_query_callback
         self.current_state = None
+        self.retries = 0
+        self.current_timeout = 0
+        self.transition_lock = Lock()
+        self._reset_state_machine()
+
+    def _increment_retries(self):
+        if self.retries < self.MAX_RADIUS_RETRIES:
+            self.retries += 1
+
+    def _state_transition(self, target, expected=None):
+        if expected is not None:
+            message = 'state was %s expected %s' % (self.current_state, expected)
+            assert self.current_state == expected, message
+        LOGGER.info('Transition: %s -> %s', self.current_state, target)
+        self.current_state = target
+
+    def _reset_state_machine(self):
+        self._state_transition(self.UNAUTH)
+        self.retries = 0
+        self.current_timeout = time.time() + self.REJECT_STATE_TIMEOUT
+        self.auth_callback(self.src_mac)
```

Comment:
this needs more than default arguments to indicate failure -- I would suggest passing None/None or whatever to clearly indicate what is going on. Currently, it looks like it's defaulting to some kind of success situation.